utils/src/pickTreeItem/quickPickAzureResource/QuickPickAzureSubscriptionStep.ts

```diff
@@ -30,6 +34,14 @@ export class QuickPickAzureSubscriptionStep extends GenericQuickPickStepWithComm
 
         return pickedSubscription;
     }
+
+    protected override getPickWithoutPrompt(picks: IAzureQuickPickItem<SubscriptionItem>[]): IAzureQuickPickItem<SubscriptionItem> | undefined {

```

Comment:
Instead of this implementation I'd try to do a skip-if-one step with pick filters that select only the chosen subscription.